### DISCUSSION TITLE
L245

### DISCUSSION THREAD

**Discussion by kwakeham** [Ralim/IronOS#1516] [Q&A]
The L245 uses JBC C245 cartridge's but it looks sort of like a TS100 / TS80 with a simpler case and buttons moved but I don't have one yet to find out.

SDG did a video recently and the lower resistance of the real JBC tips made it not work but it says it works if you provide straight DC.

Has anyone looked into if the L245 could be compatible and resolve these issues with USB-C PD?

**Comment by Ralim** [comment]
I'm not following what this has to do with IronOS?
I dont even know what this L245  _is_.
Is this a new open source soldering iron?
What issues are you talking about with USB-PD?

**Reply by kwakeham** [reply]
L245 is another new Iron yes. It seems to, but unconfirmed, to have the same or similar architecture / pcb to TS101. 

Its big advantage is it can fit real C245 JBC tips connection but is supplied by USB-C PD. The clone tips are higher resistance than real ones and when a real JBC tip is used in the L245 soldering iron, it tries to pull too much current and trips out USB-C PD supplies.

Marco Reps produced a video where he modified the IronOS code to make a TS100 have a reduced maximum duty cycle enabling him to accept JBC tips with a modified connection.

What I'm wondering if anyone has checked compatibility yet, and if Marco Reps same method of reducing duty cycle could enable the L245 to not trip out over current protection on USB-C PD supplies.

**Comment by ithinkido** [comment]
https://www.youtube.com/watch?v=b-jEbLaf93M

**Comment by MaikPfaffenrath** [comment]
The idea of a USB-C iron with genuine JBC tips is very cool in my opinion. I for example have got a big JBC setup with six irons at my workbench but I need a small and lightweight USB-C iron for my toolkit to go on installation jobs. The JBC tips work again in my opinion much better compared to the TS-100 tips and are available in a huge variety, plus they can be bought used if you want to safe money.  But as SDGs YouTube video demonstrates, they did a rather terrible job with the firmware and instead of proper PWM power management they produced fake JBC tips with a higher resistance in order to avoid tripping the USB-PD 5A current limit. That’s were ironOS could come in very handy!
But sadly it's not opencource and the chip designators are sanded off :(
I would happily send you a L245 iron (plus a genuine tip) for free (or the money so you can buy it yourself) if you are interested in porting ironOS for it!

**Reply by alextrical** [reply]
Not sure if you are still interested, but I'm planning on making a mod for the Pinecil V2 to support these tips, are you still offering a tip towards getting support?

**Reply by MaikPfaffenrath** [reply]
Yes, I am also interested in this idea! Just contact me for adress details. maik.pfaffenrath (at) posteo.de

**Comment by espenmartin** [comment]
bump to this thread and I'm also willing to help out where I can to get ironOS on the L245. Just bought one and think this would be a great addition to USB-C iron as it fills a niche that pine64 and miniware don't.

**Reply by alextrical** [reply]
Do you have a ST-Link device? If so you may be able to use the 2 exposed pads connected to PA13 and PA14 to determine the MCU family being used.
Unfortunately it's likely a similar spec to the PTS100 T12 iron I have, with very low ram, manybe down to 4kb
It may be possible to replace the chip used with a higher spec TSSOP20 STM32 chip, though they max out at 16Kb ram in that package type

**Comment by jonasius** [comment]
I just checked if the µC is connected to the USB-C DP and DN Pins. If I made no mistake, there is no connection. So at least the µC isn't capable to be updated via the USB-C connector. 
There are 4 pads on the underside of the PCB, two of them are marked with 3V3 and GND. Didn't checked them yet.

This is a picture of the scraped µC. It seems that there was a white bar on the top (of the image) before remove of the marking.
![image](https://user-images.githubusercontent.com/5968090/227795120-0285ab0d-babf-4a6f-811e-47661e24c887.png)

**Reply by alextrical** [reply]
From what I can guess from a similar iron that uses T12 bits and goes by the name PTS100
That chip is likely something along the lines of a 32F030x6
Usually a STM32 clone, in a TSSOP20 package, you could check the pinout to confirm the 2 programming pads go to PA13 & PA14.

The issue however will likely be the ram available on the chip. As the TSSOP20 is seemingly only used by the budget range. Ram is at best 16K. And at worst 4K
IronOS expects 20K of ram.

A better solution would be a open source iron handle or soldering station, that I'm happy to design. Pop a message in the [C245 discussion](https://github.com/Ralim/IronOS/discussions/1693) if you are interested

**Comment by alextrical** [comment]
If an open source C245 iron would be of use to you, pop a message in the other channel, if there is enough demand I will start getting something designed for these tips. I'm liking the idea of a 3s warmup on a 100W USB-C PSU ;)
I also know an industrial designer who worked for Roomba, and has a Dyson award under his belt, who would be happy to design a printable enclosure for it 😁

**Comment by alextrical** [comment]
Anyone here have a picture of the inside of a L245 I'm interested in how they make electrical contact with the tip so that I can design an adapter for the Pinecil V2
